src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -2181,12 +2181,26 @@ static void Main()
     static string M2(string x) => throw null!;
 }";
 
-            // missing warnings
-            // Tracked by https://github.com/dotnet/roslyn/issues/32698
-
             var comp = CreateCompilation(source, options: WithNonNullTypesTrue(TestOptions.DebugExe));
             comp.VerifyDiagnostics(
-                );
+                // (5,69): warning CS8603: Possible null reference return.
+                //         _ = new System.Func<string, string>((string? x) => { return null; }!);
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(5, 69),
+                // (11,57): warning CS8622: Nullability of reference types in type of parameter 'x' of 'lambda expression' doesn't match the target delegate 'Func<string, string>'.
```

Comment:
>// (11,57): warning CS8622: Nullability of reference types in type of parameter 'x' of 'lambda expression' doesn't match the target delegate 'Func<string, string>'. [](start = 16, length = 164)

It feels like this warning is unexpected.

Reply:
The behavior is consistent with the implicit conversion, and with the similar conversion behavior when using base and derived types:
```
System.Func<string, string> f =
    (string? x) => { return null; }; // warning
_ = new System.Func<string, string>(
    (object x) => { return ""; }); // warning
```
We can relax the requirement that parameter nullability matches but I'd prefer to do so in a separate PR.

---
In reply to: [281751757](https://github.com/dotnet/roslyn/pull/35547#discussion_r281751757) [](ancestors = 281751757)

Reply:
Logged [https://github.com/dotnet/roslyn/issues/35564](https://github.com/dotnet/roslyn/issues/35564)

---
In reply to: [281757577](https://github.com/dotnet/roslyn/pull/35547#discussion_r281757577) [](ancestors = 281757577,281751757)